src/sentry/demo/data_population.py

```diff
@@ -87,55 +105,264 @@ def generate_user():
+    Evenly spaces out all breadcrumbs starting at BREADCRUMB_LOOKBACK_TIME ago
+    """
+    breadcrumbs = event_json.get("breadcrumbs", {}).get("values", [])
+    num_breadcrumbs = len(breadcrumbs)
+    breadcrumb_time_step = BREADCRUMB_LOOKBACK_TIME * 1.0 / num_breadcrumbs
+
+    curr_time = event_json["timestamp"] - timedelta(seconds=BREADCRUMB_LOOKBACK_TIME)
+    for breadcrumb in breadcrumbs:
+        breadcrumb["timestamp"] = curr_time.replace(tzinfo=pytz.utc).timestamp()
+        curr_time += timedelta(seconds=breadcrumb_time_step)
+
+
+def populate_connected_event_scenario_1(react_project: Project, python_project: Project):
+    """
+    This function populates a set of four related events with the same trace id:
+    - Front-end transaction
+    - Front-end error
+    - Back-end transaction
+    - Back-end error
+    Occurrance times and durations are randomized
+    """
+    react_transaction = get_event_from_file("src/sentry/demo/data/react_transaction_1.json")
+    react_error = get_event_from_file("src/sentry/demo/data/react_error_1.json")
+    python_transaction = get_event_from_file("src/sentry/demo/data/python_transaction_1.json")
+    python_error = get_event_from_file("src/sentry/demo/data/python_error_1.json")
 
     for day in range(MAX_DAYS):
         for hour in range(24):
-            base = dist_function(hour)
+            base = distribution_v1(hour)
             # determine the number of events we want in this hour
             num_events = int((BASE_OFFSET + SCALE_FACTOR * base) * random.uniform(0.6, 1.0))
             for i in range(num_events):
-                # pick the minutes randomly (which means events will received be out of order)
+                # pick the minutes randomly (which means events will sent be out of order)
                 minute = random.randint(0, 60)
                 timestamp = timezone.now() - timedelta(days=day, hours=hour, minutes=minute)
-                local_error = error_json.copy()
-                local_error.update(
-                    project=project,
-                    platform=project.platform,
+                timestamp = timestamp.replace(tzinfo=pytz.utc)
+                transaction_user = generate_user()
+                trace_id = uuid4().hex
+
+                old_span_id = react_transaction["contexts"]["trace"]["span_id"]
+                frontend_root_span_id = uuid4().hex[:16]
+                frontend_duration = random_normal(2000 - 50 * day, 250, 1000) / 1000.0
+
+                frontend_context = {
+                    "trace": {
+                        "type": "trace",
+                        "trace_id": trace_id,
+                        "span_id": frontend_root_span_id,
+                    }
+                }
+
+                # React transaction
+                local_event = copy.deepcopy(react_transaction)
+                local_event.update(
+                    project=react_project,
+                    platform="javascript-transaction",
```

Comment:
@wmak So the problem with that is it causes an error because we do some automatic lookup based on the file. Passing in `react` or `react-transaction` actually breaks things because there is no sample file of that name. Maybe I need to invest into fixing this. Though unwinding some of this sample code is proving difficult...

Reply:
updated so we can have the right platform